core/dbt/events/contextvars.py

```diff
@@ -0,0 +1,84 @@
+import contextlib
```

Comment:
I'm not entirely convinced we need a general mechanism for context variables in logging, rather than just managing a single context variable for node_info. Do you think we'll have other uses for this sub-module?

Reply:
Almost all of the functions in this package are actually used.

Reply:
Sorry, it was confusing of me to attach the comment to this line. I meant that the mechanism implemented by this entire file, contextvars.py, seems more general that we need for node_info support, and I was wondering if you thought we would use it for other logging related context variables in the future.

Reply:
Yes, originally I thought there would be multiple things that we need to provide, and it's still possible that other sorts of enrichment will come up. I did look at providing unique_id to a different set of events, but it turned out that they didn't actually need it. 

All of the code is used, pretty much. In the current implementation we could collapse a couple of functions (get_contextvars and get_node_info), but I think the current factoring makes more clear what's happening. And if other things we want to do this way come up, the code is in place.